modules/TaskDetailsPage/TaskDetails.tsx

```diff
@@ -0,0 +1,200 @@
+
+import classes from "./TaskDetails.module.css";
+import { Task } from "../HomePage/types/Task";
+import useTaskOperations from "../../shared/utils/usetaskOperations";
+
+const TaskDetails = () => {
+  const router = useRouter();
+  const { taskId } = router.query;
+
+  const { getTaskById, saveTask, deleteTask, toggleCompletedTask } =
+    useTaskOperations();
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [editFormState, setEditFormState] = useState({
+    title: "",
+    description: "",
+    dueDate: "",
+    priority: 1,
+  });
+
+  const task =
+    taskId && typeof taskId === "string" ? getTaskById(taskId) : null;
+
+  useEffect(() => {
+    if (task) {
+      setEditFormState({
+        title: task.title,
+        description: task.summary,
+        dueDate: task.dueDate.toISOString().split("T")[0],
+        priority: task.priority,
+      });
+    }
+  }, [task]);
+
+  const handleEditChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    const { name, value } = e.target;
+    setEditFormState((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const handleEditSubmit = () => {
+    if (task) {
+      const updatedTask: Task = {
+        ...task,
+        title: editFormState.title,
+        summary: editFormState.description,
+        dueDate: new Date(editFormState.dueDate),
+        priority: editFormState.priority,
+      };
+      saveTask(updatedTask);
+      setIsEditing(false);
+    }
+  };
+
+  const handleDelete = () => {
+    if (task) {
+      deleteTask(task.id);
+      router.push("/todo");
+    }
+  };
+
+  const handleComplete = () => {
+    if (task) {
+      toggleCompletedTask(task.id);
+    }
+  };
+
+  if (!task) return <p>Task not found</p>;
+
+  return (
+    <Container className={classes.taskContainer}>
+      {isEditing ? (
+        <Paper p="md" style={{ maxWidth: 600, margin: "0 auto" }}>
+          <TextInput
+            name="title"
+            value={editFormState.title}
+            onChange={handleEditChange}
+            placeholder="Title"
+            label="Title"
+          />
+          <TextInput
+            name="description"
+            value={editFormState.description}
+            onChange={handleEditChange}
+            placeholder="Description"
+            label="Description"
+            mt="md"
+          />
+          <TextInput
+            type="date"
+            name="dueDate"
+            value={editFormState.dueDate}
+            onChange={handleEditChange}
+            label="Due Date"
+            mt="md"
+          />
+          <Select
+            name="priority"
+            value={editFormState.priority.toString()}
+            onChange={(value) => {
+              if (value) {
+                setEditFormState((prev) => ({
+                  ...prev,
+                  priority: parseInt(value, 10),
+                }));
+              }
+            }}
+            data={[
+              { value: "1", label: "High" },
```

Comment:
define a constant for this options (for e.g 
```
export const PRIORITY_OPTIONS = [
{},
{},
{}
];
```

) in constants file and then use it here. Look into nextJS file/folder structure doc on where to define the constants file.